gtsam/slam/tests/testEssentialMatrixFactor.cpp

```diff
@@ -526,7 +633,59 @@ TEST (EssentialMatrixFactor3, extraTest) {
   }
 }
 
-} // namespace example2
+TEST(EssentialMatrixFactor4, extraMinimization) {
+  // Additional test with camera moving in positive X direction
+
+  NonlinearFactorGraph graph;
+  for (size_t i = 0; i < 5; i++)
+    graph.emplace_shared<EssentialMatrixFactor4<Cal3Bundler>>(1, 2, pA(i),
+                                                              pB(i), model1);
+
+  // Check error at ground truth
+  Values truth;
+  truth.insert(1, trueE);
+  truth.insert(2, trueK);
+  EXPECT_DOUBLES_EQUAL(0, graph.error(truth), 1e-8);
+
+  // Check error at initial estimate
+  Values initial;
+  EssentialMatrix initialE =
+      trueE.retract((Vector(5) << 0.1, -0.1, 0.1, 0.1, -0.1).finished());
+  Cal3Bundler initialK =
+      trueK.retract((Vector(3) << 0.1, -0.02, 0.03).finished());
```

Comment:
Did you get these values from elsewhere in gtsam? I wonder if the convergence problem is because it is too far from trueK. Even 0.02 and 0.03 might be too much for distortion parameters. The tolerance used in calibrate() is 1e-5.  I'm not sure about this though.

Reply:
And 0.1 is too small for focal length, which has a true value of 500?